src/main/java/ch/njol/skript/conditions/CondIsNamed.java

```diff
@@ -0,0 +1,93 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.SkriptConfig;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.conditions.base.PropertyCondition.PropertyType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.ExprName;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is Named")
+@Description("Checks whether or not a item, block, slot, inventory, player or entity is named or has a given name")
+@Examples({
+	"on right click with a cake:",
+	"\tif event-item is named \"&bBirthday Cake\"",
+	"\t\tsend \"Happy Birthday, %player's name%!\"",
+	"",
+	"on damage of a player:",
+	"\tattacker's tool is named",
+	"\tcancel event",
+	"",
+	"on inventory click:",
+	"\tevent-inventory is named \"Example Inventory\":",
+	"\t\tcancel event",
+	""
+})
+@Since("INSERT VERSION")
+public class CondIsNamed extends Condition {
+
+	@Nullable
+	private Expression<String> name;
+	private Expression<Object> objects;
+	private boolean caseSensitive = false;
+	private static final ExprName exprName = new ExprName();
+
+	static {
+		PropertyCondition.register(CondIsNamed.class, PropertyType.BE, "named [%-string%]", "offlineplayers/entities/blocks/itemtypes/inventories/slots/worlds");
```

Comment:
```suggestion
		PropertyCondition.register(CondIsNamed.class, "named [%-string%]", "offlineplayers/entities/blocks/itemtypes/inventories/slots/worlds");
```
No need to explicitly specify the `PropertyType` as it's `PropertyType.BE` by default